program/rust/src/time_machine_types.rs

```diff
@@ -1,4 +1,36 @@
-#[derive(Debug, Clone)]
+use crate::c_oracle_header::{
+    pc_price_t,
+    EXTRA_PUBLISHER_SPACE,
+};
+use crate::error::OracleError;
+use bytemuck::{
+    Pod,
+    Zeroable,
+};
+use solana_program::msg;
+
+pub trait Tracker {
+    /// add the first price to a zero initialized tracker
+    fn add_first_price(
```

Comment:
maybe call this `initialize`?

Reply:
sure!

Reply:
Actually, I changed the interface a little bit, lmk what you think of the new name!